### PR TITLE
feat(github-actions): add bullmq tests in daily builds

### LINKED ISSUE #1398: Cover Dragonfly with BullMq tests
Run Bullmq tests with dragonfly. Run it periodically inside daily build pipeline.

### PR DESCRIPTION
**Pull request by Abhra303**
Fixes #1398 

cc: @romange

### CODE REVIEW

**Comment by Abhra303** [comment]
Is there any way to test the pipeline now?

File: .github/workflows/daily-builds.yml [hunk 1]
```diff
@@ -53,8 +53,19 @@ jobs:
           cmake -B ${{github.workspace}}/build -DCMAKE_BUILD_TYPE=Debug -GNinja -DCMAKE_CXX_COMPILER_LAUNCHER=ccache ${{ matrix.flags }}
           cd ${{github.workspace}}/build && pwd
           du -hcs _deps/
-      - name: Build
+      - name: Build & Run
         run: |
           cd ${{github.workspace}}/build
           ninja src/all
           ccache --show-stats
+          ./dragonfly --alsologtostderr --default_lua_config=allow-undeclared-keys
```
Comment (romange) [inline .github/workflows/daily-builds.yml:61]: can you show me the log of the example run for this action?

Comment (Abhra303) [reply .github/workflows/daily-builds.yml:61]: It is changed in my recent changes - https://github.com/Abhra303/dragonfly/actions/runs/5447606193/workflow. There are still some issues regarding the installation of NodeJs. Once that fixed, I will push the newest change in this branch.

**Comment by romange** [comment]
also need `--default_lua_flags=allow-undeclared-keys`

**Comment by Abhra303** [comment]
latest CI daily build - https://github.com/Abhra303/dragonfly/actions/runs/5504113402/jobs/10029991127

File: .github/workflows/daily-builds.yml [hunk 2]
```diff
@@ -53,8 +59,20 @@ jobs:
...
           cd ${{github.workspace}}/build && pwd
           du -hcs _deps/
-      - name: Build
+      - name: Build & Run
         run: |
           cd ${{github.workspace}}/build
           ninja src/all
           ccache --show-stats
+          ./dragonfly --alsologtostderr --default_lua_flags=allow-undeclared-keys &
+      - name: Clone and build BullMQ
```
Comment (adiholden) [inline .github/workflows/daily-builds.yml:68]: please change BullMQ tests to run on release build

File: .github/workflows/daily-builds.yml [hunk 3]
```diff
@@ -53,8 +59,20 @@ jobs:
...
+        run: |
+          cd ${{github.workspace}}
+          git clone https://github.com/taskforcesh/bullmq.git
+          cd bullmq
+          yarn install --ignore-engines --frozen-lockfile --non-interactive
+          yarn build
+      - name: Test BullMQ with dragonfly
+        run: |
+          cd ${{github.workspace}}/bullmq
+          yarn test
```
Comment (adiholden) [inline .github/workflows/daily-builds.yml:78]: we want to exclude one test that we have a problem with
please run
yarn test -i  -g "should process delayed jobs with several workers respecting delay"

Comment (Abhra303) [reply .github/workflows/daily-builds.yml:78]: Should I add a new job "bullmq" then?

File: .github/workflows/daily-builds.yml (hunk 2 above)
Comment (Abhra303) [reply .github/workflows/daily-builds.yml:68]: In that case, I am adding a new job "bullmq" instead of testing bullmq in the "build" job

Comment (Abhra303) [reply .github/workflows/daily-builds.yml:68]: decided to run the tests in same job

**Comment by Abhra303** [comment]
ci build - https://github.com/Abhra303/dragonfly/actions/runs/5643749718/job/15286179589

**Review by adiholden — APPROVED** [review]

**Pull request merged**